src/order_cmd.cpp

```diff
@@ -2095,6 +2095,13 @@ bool UpdateOrderDest(Vehicle *v, const Order *order, int conditional_depth, bool
 			} else {
 				if (v->type != VEH_AIRCRAFT) {
 					v->dest_tile = Depot::Get(order->GetDestination())->xy;
+				} else {
+					Aircraft *a = Aircraft::From(v);
+					DestinationID destination = a->current_order.GetDestination();
+					if (a->targetairport != destination) {
+						/* The aircraft is now heading for a different hangar than the next in the orders */
```

Comment:
It took me a bit of trial and error to understand the issue being fixed here. Rewording it in a way I think I would understand: If aircraft is currently heading for airport A, you you click Skip Order, and the next order is go to hangar at airport B, the aircraft will continue heading to airport A and use the hangar at that airport.

Reply:
The bug is this.
1 - Go to hangar A
2 - Go to hangar B
3 - Go to hangar C

If the aircraft is executing order 1, it is going to hangar A. Now you manually skip to order 2. The bug is that the aircraft is still heading to hangar A, while it should be going to hangar B. If you skip it again, it still goes to hangar A, instead of hangar C. That is the bug.